src/EventSubscriber/HookEventBridge.php

```diff
@@ -34,19 +43,32 @@ public static function getStaticSubscribedEvents(): array
 
 	public static function onDataFilterEvent(DataFilterEvent $event): void
 	{
-		$data = $event->getData();
-
-		Hook::callAll($event->getName(), $data);
-
-		$event->setData($data);
+		$event->setData(
+			static::callHook($event->getName(), $event->getData())
+		);
 	}
 
 	public static function onHtmlFilterEvent(HtmlFilterEvent $event): void
 	{
-		$html = $event->getHtml();
+		$event->setHtml(
+			static::callHook($event->getName(), $event->getHtml())
+		);
+	}
+
+	/**
+	 * @param string|array $data
+	 *
+	 * @return string|array
+	 */
+	private static function callHook(string $name, $data)
+	{
+		// Little hack to allow mocking the Hook call in tests.
```

Comment:
Please don't shortchange yourself!
```suggestion
		// Condition allowing mocking the Hook call in tests.
```

Reply:
This code will never run in production, but only in test mode. This code should not exist. But I justify this to myself because the benefit that comes from the test (when we convert the hooks to events) is greater than this bad code. That's why this is a hack and I would like to call it a hack so that later readers of the code know that this is bad code. This hack will inevitably be deleted after the hook class is completely removed.

Reply:
You're way too good for Friendica 😊

Reply:
Thank you